From 805365980feaec626e80c6514b46a6e4de319b77 Mon Sep 17 00:00:00 2001
From: Evgeny Vereshchagin <evvers@ya.ru>
Date: Fri, 29 Apr 2016 09:40:23 +0200
Subject: [PATCH] core: fix memory leak on set-default, enable, disable etc

Cherry-picked from: 24f412ca4150b490648ab8de45c6eda5bd697fd8
Related: #1331667
---
 src/core/dbus-manager.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/dbus-manager.c b/src/core/dbus-manager.c
index faa124d92..1a5525e50 100644
--- a/src/core/dbus-manager.c
+++ b/src/core/dbus-manager.c
@@ -1621,6 +1621,7 @@ static int reply_unit_file_changes_and_free(
         if (r < 0)
                 goto fail;
 
+        unit_file_changes_free(changes, n_changes);
         return sd_bus_send(bus, reply, NULL);
 
 fail:
